File: week11/15/4_Thu_longestCommonPrefix.py
from typing import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 1:
            return strs[0]
        commonprefix = ""
        minlen = len(strs[0])        
        for i in range(1, len(strs)):
            minlen = min(minlen, len(strs[i]))      

        for j in range(minlen):
            letter = strs[0][j]
            for i in range(1, len(strs)):                                 
                if strs[i][j] != letter:
                    return commonprefix
            commonprefix = commonprefix+letter    
        return commonprefix
        

File: week11/15/4_Thu_longestCommonPrefix.cpp
#include <vector>
#include <string>
#include <algorithm>

using namespace std;

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string commonprefix = "";
        sort(strs.begin(), strs.end());
        string first = strs[0], last = strs[strs.size()-1];
        for (int i=0; i < min(first.size(), last.size()); i++){
            if (first[i] != last[i]) return commonprefix;
            commonprefix += first[i];    
        }
        return commonprefix;
    }
};